platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -40,29 +90,103 @@ public class LongListDisk extends LongList {
      * @throws IOException If there was a problem reading the file
      */
     public LongListDisk(final Path file) throws IOException {
-        super(FileChannel.open(file, StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE));
-        this.file = file;
+        this(file, DEFAULT_RESERVED_BUFFER_LENGTH);
+    }
+
+    LongListDisk(final Path file, final long reservedBufferLength) throws IOException {
+        super(file, reservedBufferLength);
+        freeChunks = new ConcurrentLinkedDeque<>();
+    }
+
+    @Override
+    protected void readBodyFromFileChannelOnInit(String sourceFileName, FileChannel fileChannel) throws IOException {
+        Path tempFile = createTempFile(sourceFileName);
+        // create temporary file for writing
+        @SuppressWarnings("resource")
+        final RandomAccessFile rf = new RandomAccessFile(tempFile.toFile(), "rw");
+        // ensure that the amount of disk space is enough
+        // two additional chunks are required to accommodate "compressed" first and last chunks in the original file
+        rf.setLength(fileChannel.size() + (long) 2 * memoryChunkSize);
```

Comment:
replace `2` with `2L` removes the need to cast to long...

Reply:
Fair enough.

Reply:
?